### DISCUSSION TITLE
debug log of prisma queries?

### DISCUSSION THREAD

**Discussion by andykais** [prisma/prisma#8873] [Q&A]
Hi, is there a command in prisma that can log exactly what queries are being executed by prisma client? E.g. I imagine that

```ts
const user = await client.user.create({ data: { name: 'John Smith' }})
```
in sqlite is actually performing
1. `INSERT INTO user (name) VALUES ('John Smith')`
2. `SELECT * FROM user`

but for more complex queries, it is harder to predict. It would be awesome if I could just add a debug flag somewhere to see what it is actually doing

**Comment by andykais** [comment]
I now see that this exists! It looks as simple as:
```ts
const prisma = new PrismaClient({log:  ['query',]})
```
```
prisma:query BEGIN
prisma:query INSERT INTO `main`.`User` (`name`, `created_at`, `updated_at`) VALUES (?,?,?) RETURNING id
prisma:query SELECT `main`.`User`.`id`, `main`.`User`.`name`, `main`.User`.`created_at`, `main`.`User`.`updated_at` FROM `main`.`User` WHERE `main`.`User`.`id` = ? LIMIT ? OFFSET ?
prisma:query COMMIT
```

**Comment by andykais** [comment]
now I can start to bother the devs about more efficient queries 😅. E.g. for this we could probably reduce it to a single query:
```
prisma:query INSERT INTO `main`.`User` (`name`, `created_at`, `updated_at`) VALUES (?,?,?) RETURNING id, name, created_at
```

**Comment by ryands17** [comment]
@andykais 👋 

We still have to work on optimisations so it would be great if you could add a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for implementing `returning` so that we can look into this.

We already use `returning` for the `id` so this must be a design decision that I am not aware of. Best open a request and we can check that out 🙂

**Reply by andykais** [reply]
linked here https://github.com/prisma/prisma/issues/8948

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
